Use SETF8/16 for 8/16-bit INC/DEC

alyssarosenzweig opened the pull request:
This isn't as useful as you'd like, since the definition of SETF seems to be broken (due to a fundamental misunderstanding at Arm about how OF works?). But it's still a win in this one case.

Sonicadvance1 reviewed (approved):
Fun little optimization. Weird overflow behaviour is weird.

alyssarosenzweig commented:
Realized on my walk tonight that I forgot to update the flag opt pass. Fixed, and added an instcountci test that would've flagged the problem (pun intended)

Sonicadvance1 reviewed (approved):
Nice catch on the oversight there
